pkg/datarepair/checker/checker.go

```diff
@@ -217,24 +206,26 @@ func (checker *Checker) invalidNodes(ctx context.Context, nodeIDs storj.NodeIDLi
 
 	for i, nID := range nodeIDs {
 		if invalidNodesMap[nID] {
-			invalidNodes = append(invalidNodes, int32(i))
+			invalidNodes = append(invalidNodes, i)
 		}
 	}
 
 	return invalidNodes, nil
 }
 
 // combine the offline nodes with nodes marked invalid by statdb
-func combineOfflineWithInvalid(offlineNodes []int32, invalidNodes []int32) (missingPieces []int32) {
-	missingPieces = append(missingPieces, offlineNodes...)
+func combineOfflineWithInvalid(offlineNodes []int, invalidNodes []int) (missingPieces []int32) {
+	for _, offline := range offlineNodes {
+		missingPieces = append(missingPieces, int32(offline))
+	}
 
-	offlineMap := make(map[int32]bool)
+	offlineMap := make(map[int]bool)
```

Comment:
why does the int type change depending on if they're in the offlineMap or missingPieces array?

Reply:
internally it's easier to use `int`-s, but since the protobuf requires specific sizes, it needs to use int32 as the result